lib/CFRPageActions.jsm

```diff
@@ -431,7 +460,18 @@ const CFRPageActions = {
     if (browser !== win.gBrowser.selectedBrowser || !isHostMatch(browser, host)) {
       return false;
     }
-    const {id, content} = recommendation;
+    const {id, content, template} = recommendation;
+
+    // If this is a CFR for an add-on fetch the url for the latest xpi
+    if (template === "cfr_doorhanger") {
+      const url = await this._fetchLatestAddonVersion(content.addon);
+      // If we failed to get a url to the latest xpi, do not show a recommendation at all
+      if (!url) {
+        return false;
+      }
+      // Update the action's data with the url to the latest xpi
+      content.buttons.primary.action.data.url = url;
```

Comment:
modifying this object directly might cause side effects so is generally not a great idea, although updating a property this deep is gonna be a LOT of `...`